### PR TITLE
[12.0][IMP] module_auto_update: UI button

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Adds UI button in developer mode under _Apps_ menu.

### CODE REVIEW

File: module_auto_update/__manifest__.py [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
     'summary': 'Automatically update Odoo modules',
-    'version': '12.0.2.0.4',
+    'version': '12.0.2.0.5',
     'category': 'Extra Tools',
     'website': 'https://github.com/OCA/server-tools',
     'author': 'LasLabs, '
               'Juan José Scarafía, '
               'Tecnativa, '
               'ACSONE SA/NV, '
+              'Brainbean Apps, '
```
Comment (pedrobaeza) [inline module_auto_update/__manifest__.py]: This doesn't bring you as co-author.

File: module_auto_update/addon_hash.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Copyright 2018 ACSONE SA/NV.
-# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (pedrobaeza) [inline module_auto_update/addon_hash.py]: License is LGPL in this case. Why you change it?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: module_auto_update/addon_hash.py (hunk 2 above)
Comment (alexey-pelykh) [reply module_auto_update/addon_hash.py]: Replace tricked me :) `http` => `https` it should be

File: module_auto_update/__manifest__.py (hunk 1 above)
Comment (alexey-pelykh) [reply module_auto_update/__manifest__.py]: Righty

**Comment by alexey-pelykh** [comment]
@pedrobaeza ready for re-review

**Review by pedrobaeza — APPROVED** [review]

**Review by tafaRU — APPROVED** [review]

**Review by yajo — COMMENTED** [review]
This is not recommended because, after an upgrade, the only safe thing to do is to reboot Odoo. Also, to upgrade Odoo, the only safe way to do it is trough CLI, or the server could hang on boot if there are i.e. new fields declared.

That being said, this is not actually breaking anything. It's just opening a path for errors. My recommendation is to close this. :-1:

**Comment by alexey-pelykh** [comment]
Nothing would stop user in debug mode to go in module form and Upgrade it manually w/o restarting, which this button automates. IMHO, concern is valid, yet irrelevant for the use-case

**Comment by alexey-pelykh** [comment]
@pedrobaeza shall this be merged?

**Comment by pedrobaeza** [comment]
@Yajo's arguments are strong...

**Comment by alexey-pelykh** [comment]
If to extrapolate those arguments, by default Upgrade button should be disabled on modules UI? What harm is automating clicking on that button using code?

**Comment by yajo** [comment]
I already explained the harm and casted my vote. However, if you get enough thumbs up from others you can still merge, but I cannot vote something I think is wrong, sorry. 🤷🏻‍♂️

**Comment by alexey-pelykh** [comment]
@Yajo absolutely, I totally understand your position!

@pedrobaeza what is the procedure in this case? How many approvals and from whom are needed?

**Comment by pedrobaeza** [comment]
Why don't you add a warning about @Yajo's concerns in the README and go this way?

File: module_auto_update/README.rst [hunk 3]
```diff
@@ -63,7 +63,10 @@ the method ``_save_installed_checksums`` can be invoked in a situation
 where one is sure all modules on disk are installed and up-to-date in the
 database.
 
-An easy way to invoke this upgrade mechanism is by issuing the following
+To invoke the upgrade mechanism, navigate to *Apps* menu and use the
+*Auto-Upgrade Modules* button, available only in developer mode.
+
+Another easy way to invoke this upgrade mechanism is by issuing the following
```
Comment (sbidoul) [inline module_auto_update/README.rst:70]: Please add a note saying that a server restart is highly recommended to take new python code into account.

Comment (alexey-pelykh) [reply module_auto_update/README.rst:70]: Done

**Comment by alexey-pelykh** [comment]
@pedrobaeza added, please check if the wording is proper.

**Pull request merged**